python/dgl/graphbolt/impl/neighbor_sampler.py

```diff
@@ -2,14 +2,206 @@
 
 import torch
 from torch.utils.data import functional_datapipe
+from torchdata.datapipes.iter import IterDataPipe, Mapper
 
 from ..internal import compact_csc_format, unique_and_compact_csc_formats
 
 from ..subgraph_sampler import SubgraphSampler
 from .sampled_subgraph_impl import SampledSubgraphImpl
 
 
-__all__ = ["NeighborSampler", "LayerNeighborSampler"]
+__all__ = ["NeighborSampler", "LayerNeighborSampler", "NeighborSampler2"]
+
+
+@functional_datapipe("sample_per_layer")
+class SamplePerLayer(Mapper):
+    """Sample neighbor edges from a graph for a single layer."""
+
+    def __init__(self, datapipe, sampler, fanout, replace, prob_name):
+        super().__init__(datapipe, self._sample_per_layer)
+        self.sampler = sampler
+        self.fanout = fanout
+        self.replace = replace
+        self.prob_name = prob_name
+
+    def _sample_per_layer(self, minibatch):
+        return (
```

Comment:
Why not put the result into `minibatch`? The definition of `minibatch` is to connecting different datapipes.

Reply:
Yeah, just did.